aws/table_aws_ec2_reserved_instance.go

```diff
@@ -0,0 +1,271 @@
+				Description: "A list of tags assigned to the reserved instance.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ReservedInstancesId"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(getEc2ReservedInstanceTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEc2ReservedInstanceARN,
+				Transform:   transform.FromValue().Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listEc2ReservedInstances(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	region := d.KeyColumnQualString(matrixKeyRegion)
+	plugin.Logger(ctx).Trace("listEc2ReservedInstances", "AWS_REGION", region)
+
+	// Create Session
+	svc, err := Ec2Service(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	param := &ec2.DescribeReservedInstancesInput{}
+	// List call
+	result, err := svc.DescribeReservedInstances(param)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, reservedInstance := range result.ReservedInstances {
+		d.StreamListItem(ctx, reservedInstance)
+	}
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getEc2ReservedInstance(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getEc2ReservedInstance")
+
+	region := d.KeyColumnQualString(matrixKeyRegion)
+	instanceID := d.KeyColumnQuals["reserved_instance_id"].GetStringValue()
+
+	// create service
+	svc, err := Ec2Service(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	params := &ec2.DescribeReservedInstancesInput{
+		ReservedInstancesIds: []*string{aws.String(instanceID)},
+	}
+
+	op, err := svc.DescribeReservedInstances(params)
+	if err != nil {
+		return nil, err
+	}
+
+	if op.ReservedInstances != nil && len(op.ReservedInstances) > 0 {
+		return op.ReservedInstances[0], nil
+	}
+	return nil, nil
+}
+
+func getEc2ReservedInstanceARN(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getEc2ReservedInstanceARN")
+	instance := h.Item.(*ec2.ReservedInstances)
+	region := d.KeyColumnQualString(matrixKeyRegion)
+
+	getCommonColumnsCached := plugin.HydrateFunc(getCommonColumns).WithCache()
+	commonData, err := getCommonColumnsCached(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+	commonColumnData := commonData.(*awsCommonColumnData)
+
+	arn := "arn:" + commonColumnData.Partition + ":ec2:" + region + ":" + commonColumnData.AccountId + ":instance/" + *instance.ReservedInstancesId
```

Comment:
Arn should be `arn:${Partition}:ec2:${Region}:${Account}:reserved-instances/${ReservationId}`, please check?